yew/src/virtual_dom/vtag.rs

```diff
@@ -261,36 +267,132 @@ impl VTag {
         }
     }
 
-    /// This handles patching of attributes when the keys are equal but
-    /// the values are different.
-    fn diff_attributes<'a>(
-        &'a self,
-        ancestor: &'a Option<Box<Self>>,
-    ) -> impl Iterator<Item = Patch<&'a str, &'a str>> + 'a {
-        // Only change what is necessary.
-        let to_add_or_replace =
-            self.attributes.iter().filter_map(move |(key, value)| {
-                match ancestor
-                    .as_ref()
-                    .and_then(|ancestor| ancestor.attributes.get(&**key))
-                {
-                    None => Some(Patch::Add(&**key, &**value)),
-                    Some(ancestor_value) if value != ancestor_value => {
-                        Some(Patch::Replace(&**key, &**value))
+    /// Diffs attributes between the old and new lists.
+    ///
+    /// This method optimises for the lists being the same length, having the
+    /// same keys in the same order - most common case.
+    fn diff_attributes_vectors<'a>(
```

Comment:
This could be more efficient in the case where an attribute is appended. Instead of short-circuiting on `new.len() != old.len()` we can iterate over all common attributes before calling `diff_incompatible` at the end to handle the appended attributes. This is how `diff_attributes_indexmaps` works.

Reply:
I've tried implementing this with peekable iterators before, but Godbolt showed to generate a lot more instructions for that method. So peekable iterators do optimise appends at the cost of reducing equal length vector comparison speed - arguably the most common case.

Reply:
Ah good to know, that's fine then